src/legacy/core_plugins/kibana/public/dashboard/np_ready/legacy_app.js

```diff
@@ -62,6 +64,14 @@ export function initDashboardApp(app, deps) {
     stateManagementConfigProvider.disable();
   });
 
+  app.factory('history', () => createHashHistory());
```

Comment:
nit: It's not a big deal, but as we are trying to get away from angular we could already do that step here and call createKbnUrlStorage directly within the controller instead of letting angular handle this. Not blocking this merge, we can also tackle that later.

Reply:
I needed to make sure that Listing page and Dashboard page are using the same instance of `kbnUrlStateStorage`. 

When we clean up angular from here, these 2 just should move back somewhere to the root of the plugin and then passed down to react components.

Reply:
Why is this required? I'm asking because the dashboard listing page and the actual dashboard page are two separate apps - if you are switching between the two it is unmounting and re-mounting the application anyway (new angular context and so on). https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/public/dashboard/plugin.ts#L128

That was necessary because of the different urls - the listing page is the `dashboards` app, the individual dashboard is the `dashboard` app.